crates/extensions/c8y_mapper_ext/src/actor.rs

```diff
@@ -375,7 +309,7 @@ impl C8yMapperBuilder {
         mqtt.connect_sink(config.topics.clone(), &box_builder.get_sender());
         let http_proxy = C8YHttpProxy::new(http);
         let timer_sender = timer.connect_client(box_builder.get_sender().sender_clone());
-        let upload_sender = uploader.connect_client(box_builder.get_sender().sender_clone());
+        let uploader = ClientMessageBox::new(uploader);
```

Comment:
We should be able to do the same for the downloader as well.

Reply:
We can, explanation why it was not done for now in comment: https://github.com/thin-edge/thin-edge.io/pull/2881#discussion_r1603267939